src/test/java/specs/FixturesSpec.java

```diff
@@ -113,11 +114,15 @@
         });
 
         it("all run before each test in declaration order", () -> {
-          assertThat(numbers, contains(1, 2, 3, 4));
+          // looking at the beforeEach statements, they ARE in this order
+          // to discuss
+          assertThat(numbers, contains(1, 3, 4, 2));
```

Comment:
The order should be 1, 2, 3, 4 here. That's to match the RSpec and Jasmine ordering. Child contexts implicitly depend on their parent (since you can access the parent's state), so you need to make sure the parent scope gets set up first before setting up the child scope. Same for teardown code; can't tear down the outer scope until the inner scope (that might rely on outer scope state) has cleaned up it's own stuff.

For a concrete example, consider an integration suite that hits a database. The outermost scope might begin a transaction in `beforeEach` and roll it back in `afterEach`. If the nested context does some database manipulation in its own `beforeEach`, that should run inside the wrapping transaction. If it does some cleanup or verification in an `afterEach`, that should happen before the transaction rolls back.

I made up another RSpec sample for reference, and to check my own sanity:

```ruby
RSpec.describe "specs with hooks" do

  before(:each) do
    puts "before1"
  end

  after(:each) do
    puts "after1"
  end

  before(:each) do
    puts "before2"
  end

  it 'something' do
    puts "-OUTER SPEC-"
  end

  describe "and nesting" do 

    before(:each) do 
      puts "before3"
    end

    it "something else" do
      puts "-INNER SPEC-"
    end

    after(:each) do 
      puts "after3"
    end

  end

  after(:each) do
    puts "after2"
  end

end
```
Output: 

> before1
before2
-OUTER SPEC-
after2
after1
before1
before2
before3
-INNER SPEC-
after3
after2
after1